src/evolution_vaccination.m

```diff
@@ -6,148 +6,127 @@ function evolution_vaccination()
+    r_n = system.reward(1,2) + system.reward(2,1) + system.reward(2,2);
+    r_n  = r_n/3;
+    
+    p = vaccination_probability_1(r_i - r_n);
+    p2 = rand;
+    if(p2 < p)
+        system.vaccinated(1,1) = not(system.vaccinated(1,1));
     end
-    if(system.state(1,n) == "S")
-       r_i = system.reward(1,n); %individual's reward
-       %neighbours' average reward
-       r_n = system.reward(1,n-1) + system.reward(2,n) + system.reward(2,n-1);
-       r_n  = r_n/3;
-       
-       p = vaccination_probability_1(r_i - r_n);
-       p2 = rand;
-       if(p2 < p)
-           system.vaccinated(1,n) = true;
-           system.state(1,n) = "R";
-           system.reward(1,n) = system.vaccinated(1,n) + r_vacc;
-       end
+    
+    r_i = system.reward(1,n); %individual's reward
+    %neighbours' average reward
+    r_n = system.reward(1,n-1) + system.reward(2,n) + system.reward(2,n-1);
+    r_n  = r_n/3;
+    
+    p = vaccination_probability_1(r_i - r_n);
+    p2 = rand;
+    if(p2 < p)
+        system.vaccinated(1,n) = not(system.vaccinated(1,n));
     end
-    if(system.state(n,1) == "S")
-       r_i = system.reward(n,1); %individual's reward
-       %neighbours' average reward
-       r_n = system.reward(n,2) + system.reward(n-1,1) + system.reward(n-1,2);
-       r_n  = r_n/3;
-       
-       p = vaccination_probability_1(r_i - r_n);
-       p2 = rand;
-       if(p2 < p)
-           system.vaccinated(n,1) = true;
-           system.state(n,1) = "R";
-           system.reward(n,1) = system.vaccinated(n,1) + r_vacc;
-       end
+    
+    r_i = system.reward(n,1); %individual's reward
+    %neighbours' average reward
+    r_n = system.reward(n,2) + system.reward(n-1,1) + system.reward(n-1,2);
+    r_n  = r_n/3;
+    
+    p = vaccination_probability_1(r_i - r_n);
+    p2 = rand;
+    if(p2 < p)
+        system.vaccinated(n,1) = not(system.vaccinated(n,1));
     end
-    if(system.state(n,n) == "S")
-       r_i = system.reward(n,n); %individual's reward
-       %neighbours' average reward
-       r_n = system.reward(n,n-1) + system.reward(n-1,n) + system.reward(n-1,n-1);
-       r_n  = r_n/3;
-       
-       p = vaccination_probability_1(r_i - r_n);
-       p2 = rand;
-       if(p2 < p)
-            system.vaccinated(n,n) = true;
-            system.state(n,n) = "R";
-            system.reward(n,n) = system.vaccinated(n,n) + r_vacc;
-       end
+    
+    r_i = system.reward(n,n); %individual's reward
+    %neighbours' average reward
+    r_n = system.reward(n,n-1) + system.reward(n-1,n) + system.reward(n-1,n-1);
+    r_n  = r_n/3;
+    
+    p = vaccination_probability_1(r_i - r_n);
+    p2 = rand;
+    if(p2 < p)
+        system.vaccinated(n,n) = not(system.vaccinated(n,n));
+    end  
+    for i = 1:n
+        for j = 1:n
+            if((system.state(i,j) == "S") && (system.vaccinated(i,j) == true))
+                system.state(i,j) = "R";
+                system.reward(i,j) = system.reward(i,j) + r_vacc;
+            end

```

Comment:
Utilise la puissance des matrices de Matlab, ce que tu proposes consomme trop de resources, essaie ceci
```suggestion
          right_ind = (system.state == “S” & system.vaccinated == true);
          system.state(right_ind) = string(ones(size(system.state(right_ind))))*”R”;   %pas sur de la syntax
          system.reward(right_ind) = system.reward(right_ind) + r_vacc*ones(size(system.reward(right_ind)));
```

Reply:
solved!

Reply:
i left the old code in comments just in case it doesn't work properly